### DISCUSSION TITLE
node.js unleash-client compatibility with unleash server?

### DISCUSSION THREAD

**Discussion by birchb1024** [Unleash/unleash#3458] [Q&A] [answered]
We're using server 3.3.4 and unleash-client 3.2.4. A colleague was having difficulties recently after the server was upgraded. I'm looking to upgrade to the latest client. . .

Is there some kind of a rule or document which tells us what client versions are compatible with various server versions?

**Comment by pransh15** [comment]
Hey @birchb1024, thanks for starting a discussion ❤️👋

We often focus on backward-compatibility since we understand that upgrading gets complex as you scale. I believe it should work fine, but I'll get @sighphyre involved in-case I'm wrong.

**Comment by sighphyre** [comment, marked as answer]
Hey @birchb1024, @pransh15  is correct here. We don't, unfortunately, have a compatibility table for client/server versions but the clients are built to be very backwards compatible so the latest version of the client should be compatible with any server in the 3.x.x/4.x.x range. That being said, version 3.x.x of the Unleash server is very old and we don't actively maintain that version range anymore. If you have a specific problem we'd be happy to help either here or on Slack
